File: frontend/src/lib/api/feedback.ts
// Feedback API client — public endpoint, no auth required.

import { apiFetch } from "./client";

export interface FeedbackCreate {
  name?: string;
  email?: string;
  role?: string;
  rating?: number;
  message: string;
}

export interface FeedbackOut {
  id: number;
  name: string | null;
  email: string | null;
  role: string | null;
  rating: number | null;
  message: string;
  created_at: string;
}

export async function submitFeedback(
  payload: FeedbackCreate,
): Promise<FeedbackOut> {
  return apiFetch<FeedbackOut>("/api/feedback", {
    method: "POST",
    body: payload,
  });
}

File: backend/app/schemas/feedback.py
"""Pydantic schemas for feedback endpoints."""

from datetime import datetime

from pydantic import BaseModel, Field


class FeedbackCreate(BaseModel):
    name: str | None = Field(default=None, max_length=120)
    email: str | None = Field(default=None, max_length=255)
    role: str | None = Field(default=None, max_length=120)
    rating: int | None = Field(default=None, ge=1, le=5)
    message: str = Field(min_length=1, max_length=5000)


class FeedbackOut(BaseModel):
    id: int
    name: str | None
    email: str | None
    role: str | None
    rating: int | None
    message: str
    created_at: datetime

    model_config = {"from_attributes": True}
